frontend/src/components/ui/data-grid/controllers/rows.ts

```diff
@@ -46,38 +56,32 @@ export class DataGridRowsController implements ReactiveController {
     }
   }
 
-  private setRowsFromItems<T extends GridItem = GridItem>(items: T[]) {
+  private setRowsFromItems(items: Item[]) {
     const rowKey = this.#host.rowKey;
 
     this.rows = new Map(
       this.#host.rowKey
-        ? items.map((item) => [
-            item[rowKey as unknown as string] as GridRowId,
-            item,
-          ])
+        ? items.map((item) => [item[rowKey as keyof Item] as GridRowId, item])
```

Comment:
(also following the previous comment)
```suggestion
        ? items.map((item) => [item[rowKey] as GridRowId, item])
```